shell/platform/android/android_surface_software.cc

```diff
@@ -80,17 +97,31 @@ bool AndroidSurfaceSoftware::PresentBackingStore(
     return false;
   }
 
-  // Pass the sk_surface buffer to the android FlutterView.
-  JNIEnv* env = fml::jni::AttachCurrentThread();
-
-  // Buffer will be copied into a Bitmap Java-side.
-  fml::jni::ScopedJavaLocalRef<jobject> direct_buffer(
-      env,
-      env->NewDirectByteBuffer(pixmap.writable_addr(), pixmap.getSize64()));
+  ANativeWindow_Buffer native_buffer;
+  ANativeWindow_lock(native_window_->handle(), &native_buffer, nullptr);
```

Comment:
The lock can fail. We need to check and handle that case (probably just return early) because we might end up writing to a junk native buffer.